### DISCUSSION TITLE
Exclude columns when exporting

### DISCUSSION THREAD

**Discussion by dhcmega** [yajra/laravel-datatables#2499] [Q&A]
### Summary of problem or feature request

Problem: Export is too slow
Reason?: I think the problem is that the action column is processed for each line
If I skip the action column (in my case with 20k records) export speed is many times faster.

### Code snippet of problem

```php
        return $dataTable->addColumn('action', 'admin.users.datatables_actions');
```

Actual work around
```php
        if($this->request()->filled('action')) {
            return $dataTable;
        }
        return $dataTable->addColumn('action', 'admin.users.datatables_actions');
```

I found this using `strace`
There was one of this chunks per row (user in this case).
This a simplified part of the strace output:

```
access("storage/framework/views/c08be0011b8c2177fd317340650742011c1b1208.php", F_OK) = 0
stat("resources/views/admin/users/datatables_actions.blade.php", {}) = 0
stat("storage/framework/views/c08be0011b8c2177fd317340650742011c1b1208.php", {}) = 0
lstat("resources/views/admin/users/datatables_actions.blade.php", {}) = 0
lstat("resources/views/admin/users", {}) = 0
lstat("resources/views/admin", {}) = 0
lstat("resources/views", {}) = 0
lstat("resources", {}) = 0
```

Even though the column is not exported, it's being processed.
I hope I got right and I am not confused about this.

Thanks!

### System details

- Operating System: Linux Ubuntu
- PHP Version: 7.3.11
- Laravel Version: 7.28.4
- Laravel-Datatables Version: 9.11.1

**Comment by yajra** [comment]
Are you using the DataTable service class? If yes, you can override `$exportColumns` property and exclude the action column.

```php
protected $exportColumns = [];
```

See https://yajrabox.com/docs/laravel-datatables/master/export-column for ref.

**Comment by dhcmega** [comment]
Hi, thanks for responding, I will check it.

**Comment by yajra** [comment]
Btw, you can also use `->exportable(false)` on you column definitions:

```php
Column::computed('action')->exportable(false)->printable(false);
```

**Reply by beingshaif01** [reply]
exportable(false) not works

**Reply by lokesh-zersys** [reply]
I had the same issue and fixed it by including  vendor/datatables/buttons.server-side.js in html

**Reply by ramkumar7792** [reply]
> I had the same issue and fixed it by including vendor/datatables/buttons.server-side.js in html

How you fixed

**Reply by JCamargo1951** [reply]
I fixed it without "exportable(false)" by using this in my html function when creating buttons: 
                             ->buttons([
                                            Button::make('excel')
                                                ->className('btn btn-success')
                                                ->exportOptions([
                                                    'columns' => ':not(:last-child)', // My case exclude the last one
                                                ]), 
                               ])

**Reply by dhcmega** [reply]
export false works for columns in my case, but not for actions column as it's rendering the buttons when exporting, debugbar shows 100s of renderings with complex permissions calculations, not needed for exporting.
